app/scripts/controllers/preferences-controller.ts

```diff
@@ -441,21 +534,21 @@ export default class PreferencesController {
   /**
    * Getter for the `useRequestQueue` property
    *
-   * @returns {boolean} whether this option is on or off.
+   * @returns whether this option is on or off.
    */
-  getUseRequestQueue() {
+  getUseRequestQueue(): boolean {
     return this.store.getState().useRequestQueue;
   }
 
   /**
    * Sets a custom label for an account
    *
    * @deprecated - Use setAccountName from the AccountsController
-   * @param {string} address - the account to set a label for
-   * @param {string} label - the custom label for the account
-   * @returns {Promise<string>}
+   * @param address - the account to set a label for
+   * @param label - the custom label for the account
+   * @returns
    */
-  async setAccountLabel(address, label) {
+  async setAccountLabel(address: string, label: string): Promise<string> {
```

Comment:
I believe this function shouldn't be async in first place, once removed the return type would be string

Reply:
Nit:
We could also consider updating the function by switching the order to check the address first. Additionally, it should throw an error if the account is not found to ensure that `account.id` passes TypeScript validation, as `account` could be undefined.
````
 setAccountLabel(address: string, label: string): string {
    if (!address) {
      throw new Error(
        `setAccountLabel requires a valid address, got ${String(address)}`,
      );
    }

    const account = this.messagingSystem.call(
      'AccountsController:getAccountByAddress',
      address,
    );

    if (!account) {
      throw new Error(`Account for '${address} not found`);
    }

    this.messagingSystem.call(
      'AccountsController:setAccountName',
      account.id,
      label,
    );

    return label;
  }
```